cmd/resourcedistributiongenerator/generator/resourcedistribution.go

```diff
@@ -77,9 +83,54 @@ type TargetsArgs struct {
 	IncludedNamespaces []string `json:"includedNamespaces,omitempty" yaml:"includedNamespaces,omitempty"`
 
 	// NamespaceLabelSelector for the generator.
-	NamespaceLabelSelector *metav1.LabelSelector `json:"namespaceLabelSelector,omitempty" yaml:"namespaceLabelSelector,omitempty"`
+	NamespaceLabelSelector *LabelSelector `json:"namespaceLabelSelector,omitempty" yaml:"namespaceLabelSelector,omitempty"`
+}
+
+// It is the same as metav1.LabelSelector except that the YAMl tag is added after each field
+//
+// A label selector is a label query over a set of resources. The result of matchLabels and
+// matchExpressions are ANDed. An empty label selector matches all objects. A null
+// label selector matches no objects.
+// +structType=atomic
+type LabelSelector struct {
+	// matchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
+	// map is equivalent to an element of matchExpressions, whose key field is "key", the
+	// operator is "In", and the values array contains only "value". The requirements are ANDed.
+	// +optional
+	MatchLabels map[string]string `json:"matchLabels,omitempty"  protobuf:"bytes,1,rep,name=matchLabels" yaml:"matchLabels,omitempty"`
+	// matchExpressions is a list of label selector requirements. The requirements are ANDed.
+	// +optional
+	MatchExpressions []LabelSelectorRequirement `json:"matchExpressions,omitempty" protobuf:"bytes,2,rep,name=matchExpressions" yaml:"matchExpressions,omitempty"`
 }
 
+// A label selector requirement is a selector that contains values, a key, and an operator that
+// relates the key and values.
+type LabelSelectorRequirement struct {
+	// key is the label key that the selector applies to.
+	// +patchMergeKey=key
+	// +patchStrategy=merge
+	Key string `json:"key" patchStrategy:"merge" patchMergeKey:"key" protobuf:"bytes,1,opt,name=key" yaml:"key,omitempty"`
+	// operator represents a key's relationship to a set of values.
+	// Valid operators are In, NotIn, Exists and DoesNotExist.
+	Operator LabelSelectorOperator `json:"operator"  protobuf:"bytes,2,opt,name=operator,casttype=LabelSelectorOperator" yaml:"operator,omitempty"`
+	// values is an array of string values. If the operator is In or NotIn,
+	// the values array must be non-empty. If the operator is Exists or DoesNotExist,
+	// the values array must be empty. This array is replaced during a strategic
+	// merge patch.
+	// +optional
+	Values []string `json:"values,omitempty" protobuf:"bytes,3,rep,name=values"  yaml:"values,omitempty"`
+}
+
+// A label selector operator is the set of operators that can be used in a selector requirement.
+type LabelSelectorOperator string
+
+const (
+	LabelSelectorOpIn           LabelSelectorOperator = "In"
```

Comment:
Just use `*metav1.LabelSelector` no need to write by ourselves  and the same with others.

Reply:
I can change LabelSelectorOperator to metav1. LabelSelectorOperator.

But, metav1.LabelSelector field don't have yaml tag, default is the field name in all lowercase.
For example, the `matchLables` field needs to be written as `matchlables` when used in kustomization, which is not the same as the API we defined.

> mostly lgtm and you can add the release/CI github actions in workflow.

i will try

Reply:
> I can change LabelSelectorOperator to metav1. LabelSelectorOperator.
> 
> But, metav1.LabelSelector field don't have yaml tag, default is the field name in all lowercase. For example, the `matchLables` field needs to be written as `matchlables` when used in kustomization, which is not the same as the API we defined.
> 
> > mostly lgtm and you can add the release/CI github actions in workflow.
> 
> i will try

Got it!